src/db/dbConnectionProvider.ts

```diff
@@ -0,0 +1,16 @@
+import mongoose from 'mongoose'
+
+const connectToDatabase = (): Promise<void> => {
+    return new Promise((resolve, reject) => {
+        if (!process.env.DB_URL) return reject()
+        mongoose.connect(process.env.DB_URL, { useNewUrlParser: true, useUnifiedTopology: true, useCreateIndex: true })
```

Comment:
Why do you encapsulate it inside a promise if it already returns then and catch?